src/lib/utils/socket/socket.cpp

```diff
@@ -46,20 +46,19 @@ class Asio_Socket final : public OS::Socket {
    public:
       Asio_Socket(std::string_view hostname, std::string_view service, std::chrono::milliseconds timeout) :
             m_timeout(timeout), m_timer(m_io), m_tcp(m_io) {
-         m_timer.expires_from_now(m_timeout);
+         m_timer.expires_after(m_timeout);
          check_timeout();
 
          boost::asio::ip::tcp::resolver resolver(m_io);
-         boost::asio::ip::tcp::resolver::query query(std::string{hostname}, std::string{service});
-         boost::asio::ip::tcp::resolver::iterator dns_iter = resolver.resolve(query);
+         boost::asio::ip::tcp::resolver::results_type dns_iter =
+            resolver.resolve(std::string{hostname}, std::string{service});
```

Comment:
Nit: This could now be calld `dns_result`, `endpoints` or something similar, as it is not technically an iterator anymore.

Reply:
A lot of this could *really* use some `auto` treatment, `boost::asio::ip::tcp::resolver::results_type` whyyyyyy